prerequisite_changes.py

```diff
@@ -0,0 +1,166 @@
+            newlines.append(line.replace('Authorization=Bearer ${jwt_token}',
+                                         'X-Auth-Token=${jwt_token}'))
+    with open(file_path, 'w') as f:
+        for line in newlines:
+            f.write(line)
+
+
+def add_data(file_path, find, new_line):
+    """
+    will do below action
+        1. will put specific line (i.e. new_line) in file.
+    """
+    with open(file_path, "r") as in_file:
+        buf = in_file.readlines()
+    with open(file_path, "w") as out_file:
+        for line in buf:
+            if find in line:
+                line = line + new_line
+            out_file.write(line)
+
+
+def change_port(old_port, new_port):
+    """
+    Will replace port in global_variables.py file
+    """
+    with open('TAF/config/global_variables.py', 'rt') as f:
+        newlines = []
+        for line in f.readlines():
+            newlines.append(line.replace(old_port, new_port))
+    with open('TAF/config/global_variables.py', 'w') as f:
+        for line in newlines:
+            f.write(line)
+
+
+def replace(file_path, old_str, new_str):
+    with open(file_path, 'rt') as f:
+        newlines = []
+        for line in f.readlines():
+            newlines.append(line.replace(old_str, new_str))
+    with open(file_path, 'w') as f:
+        for line in newlines:
+            f.write(line)
+
+
+def general_changes():
+    """
+    Note: This function will do below task
+        1. will modify "BASE_URL", "DEPLOY_TYPE" and "PORT <core-metadata> "
+        2. will change Auth type in common files
+        3. set IP address of qemu to "BASE_URL"
+    """
+    global_variable("TAF/config/global_variables.py")
+    change_auth_type("TAF/testCaseModules/keywords/setup/startup_checker.py")
+    change_auth_type("TAF/testCaseModules/keywords/common/commonKeywords.robot")
+    add_data("TAF/testCaseModules/keywords/setup/startup_checker.py",
+             "import subprocess", "import os" + "\n")
+    add_data("TAF/testCaseModules/keywords/setup/startup_checker.py",
+             "def check_service_startup(d, token):",
+             "    SettingsInfo().constant.BASE_URL=open(os.getenv(\"DATA_FILE\"), \"r\").read()\n    SettingsInfo().constant.BASE_URL=SettingsInfo("
+             ").constant.BASE_URL[:-1]" + "\n")
+    if module == "KVM" or module == "kvm":
+        add_data("TAF/testCaseModules/keywords/common/commonKeywords.robot",
+                 "Set suite variable  ${response}  ${resp.status_code}",
+                 "    Set suite variable  ${body}  ${resp.json()}\n")
+
+
+def main():
+    general_changes()
+    if service == "core-metadata":
+        # This is core-metadata service specific changes
+        change_auth_type("TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot")
+        change_port("8443/core-metadata", "8002/metadata")
+        if module == "KVM" or module == "kvm":
+            add_data("TAF/testScenarios/functionalTest/V2-API/core-metadata/device/GET-Positive.robot",
```

Comment:
Could we get a comment on each of these changes on why the change is needed?  A short, single sentence should help.  This will be important for the future if EdgeX-TAF changes anything.

Reply:
ok sure